sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -910,4 +910,45 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
       }
     }
   }
+
+  test("SPARK-35686: error out for creating view with auto gen alias") {
+    withView("v") {
+      val e = intercept[AnalysisException] {
+        sql("CREATE VIEW v AS SELECT count(*) FROM VALUES (1), (2), (3) t(a)")
+      }
+      assert(e.getMessage.contains("without explicitly assigning an alias"))
+      sql("CREATE VIEW v AS SELECT count(*) AS cnt FROM VALUES (1), (2), (3) t(a)")
+      checkAnswer(sql("SELECT * FROM v"), Seq(Row(3)))
+    }
+  }
+
+  test("SPARK-35686: error out for creating view with auto gen alias - nested type") {
+    withView("v") {
+      val e = intercept[AnalysisException] {
+        sql("CREATE VIEW v AS SELECT array(1, 2, 3)[0]")
```

Comment:
This is not a very interesting case. How about testing `SELECT * FROM (SELECT a + b ...)`?